src/dmd/dtoh.d

```diff
@@ -530,6 +530,15 @@ public:
             printf("[AST.Import enter] %s\n", i.toChars());
             scope(exit) printf("[AST.Import exit] %s\n", i.toChars());
         }
+
+        // Export aliased imports as #define
+        // Ideally should enforce some sort of module/aggregate scope
+        foreach (const idx, const name; i.names)
+        {
+            auto alias_ = i.aliases[idx];
+            if (!alias_) continue;
+            buf.printf("#define %s %s\n\n", alias_.toChars(), name.toChars());
+        }
```

Comment:
Is this code going to be run for nested aliased imports also?

Reply:
Can you provide an example?

Reply:
```d
extern (C++) struct A
{
    import std.stdio : writeln = print;
    void func();
}

void func(T)(T)
{
      import std.stdio : writeln = print;
}
```

Reply:
Yes. The current solution has the drawback that macros ignore the scope unless the header generator inserts `#undef` at the end of each scope.

Reply:
Another solution would be to remember the aliased imports and print the imported symbol without the alias.

Reply:
The problem here is that local imports are going to be translated to global defines. This may lead to all kinds of problems where the define name may clash with other symbols (coming from the same module or from a different one). I think that the best solution is a mixture of both: for global aliased imports it is fine to use a `#define` (the D compiler makes sure that there will be no duplicates), however, for scoped ones we can simply use the imported symbol (that way we avoid potential clashes with global symbols).

Reply:
Reworked the entire PR to only consider top-level imports and only emit aliases which can be translated to `using ...`